File: Frontend/src/types/equipment.ts
export type EquipmentStatus =
  | "AVAILABLE"
  | "IN_USE"
  | "UNDER_MAINTENANCE"
  | "OUT_OF_SERVICE"
  | "DECOMMISSIONED";

export const EQUIPMENT_STATUS_LABEL: Record<EquipmentStatus, string> = {
  AVAILABLE: "Disponible",
  IN_USE: "En uso",
  UNDER_MAINTENANCE: "En mantenimiento",
  OUT_OF_SERVICE: "Fuera de servicio",
  DECOMMISSIONED: "Dado de baja",
};

export interface Equipment {
  id: string;
  internalCode: string;
  serialNumber: string;
  type: string;
  brand: string;
  model: string;
  characteristics?: Record<string, unknown> | null;
  acquisitionDate?: string | null;
  acquisitionPrice?: number | null;
  warrantyUntil?: string | null;
  location: string;
  status: EquipmentStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateEquipmentRequest {
  internalCode: string;
  serialNumber: string;
  type: string;
  brand: string;
  model: string;
  characteristics?: Record<string, unknown>;
  acquisitionDate?: string;
  acquisitionPrice?: number;
  warrantyUntil?: string;
  location: string;
  status?: EquipmentStatus;
}

export type UpdateEquipmentRequest = Partial<CreateEquipmentRequest>;

File: src/Backend.API/Models/Equipment.cs
namespace Backend.API.Models;

public class Equipment
{
    public Guid Id { get; set; }
    public string InternalCode { get; set; } = string.Empty;
    public string SerialNumber { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Brand { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public string? Characteristics { get; set; }
    public DateOnly? AcquisitionDate { get; set; }
    public decimal? AcquisitionPrice { get; set; }
    public DateOnly? WarrantyUntil { get; set; }
    public string Location { get; set; } = string.Empty;
    public EquipmentStatus Status { get; set; } = EquipmentStatus.AVAILABLE;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}